scripts/generate-global-changelog.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * SPDX-FileCopyrightText: © 2021 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const {execSync} = require('child_process');
+const fs = require('fs');
+const path = require('path');
+
+const IGNORE_LIST = [
+	'packages/demos/CHANGELOG.md',
+	'clayui.com/CHANGELOG.md',
+	'CHANGELOG.md',
+];
+
+const dateObj = new Date();
+const month = dateObj.getMonth() + 1;
+const day = dateObj.getDate();
+const year = dateObj.getFullYear();
+
+const dateString = `${year}-${month}-${day}`;
+
+const globalChangeLog = path.join(__dirname, '../CHANGELOG.md');
+
+function main() {
+	const diffOutput = execSync(
+		'git diff HEAD^ --name-only -- "*CHANGELOG.md"'
+	).toString();
+
+	const changelogContents = fs.readFileSync(globalChangeLog).toString();
+
+	if (changelogContents.includes(`## ${dateString}`)) {
+		console.error(
+			`CHANGELOG has already been generated for the ${dateString} release`
+		);
+
+		return;
+	}
+
+	const files = diffOutput
+		.split(/\n/)
```

Comment:
Obviously not going to cause problems here, but as a matter of best practice, if you're going to be scripting with the output of Git commands you should use `NUL`-delimited output formats whenever they're available (and they usually are); that way, even if a filename contained a newline, you'd still parse the output correctly.

ie. add `-z` to your `git diff` invocation above, and instead `.split('\0')`